### PR TITLE
Bring merge tasks up to par with deletion logging.

### PR DESCRIPTION
**Pull request by JTCunning**
@mattrobenolt @getsentry/infrastructure 

I think I did this right.

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4129)

### CODE REVIEW

File: src/sentry/tasks/merge.py [hunk 1]
```diff
@@ -96,7 +116,7 @@ def merge_group(from_object_id=None, to_object_id=None, **kwargs):
 @instrumented_task(name='sentry.tasks.merge.rehash_group_events', queue='merge',
                    default_retry_delay=60 * 5, max_retries=None)
 @retry
-def rehash_group_events(group_id, **kwargs):
+def rehash_group_events(group_id, transaction_id=None, **kwargs):
```
Comment (JTCunning) [inline src/sentry/tasks/merge.py]: I couldn't actually find anywhere that we called this so.. shrug.

File: src/sentry/tasks/merge.py [hunk 2]
```diff
@@ -163,17 +189,17 @@ def _rehash_group_events(group, limit=100):
 
 
 def merge_objects(models, group, new_group, limit=1000,
-                  logger=None):
+                  logger=None, first_pass=True, transaction_id=None):
     from sentry.models import GroupTagKey, GroupTagValue
 
     has_more = False
+    if first_pass:
+        logger.info('objects.merge.executed', extra={
```
Comment (mattrobenolt) [inline src/sentry/tasks/merge.py]: The bad part about this is it now literally says nothing. It doesnt' say what objects, it just repeats the fact that these two groups are merging. So I'd vote to just remove this, or move it outside into `merge_group`.

File: src/sentry/tasks/merge.py [hunk 3]
```diff
@@ -220,7 +246,14 @@ def merge_objects(models, group, new_group, limit=1000,
                 except DataError:
                     # it's possible to hit an out of range value for counters
                     pass
+                obj_id = obj.id
                 obj.delete()
+                if logger is not None:
+                    delete_logger.debug('object.delete.executed', extra={
```
Comment (mattrobenolt) [inline src/sentry/tasks/merge.py]: Good call on `DEBUG` level. This will be `a lot` of data.

**Comment by mattrobenolt** [comment]
Reviewed 1 of 3 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4129)*

**Pull request merged**